cassovary-core/src/main/scala/com/twitter/cassovary/util/SortedSeq.scala

```diff
@@ -0,0 +1,114 @@
+package com.twitter.cassovary.util
+
+import java.util
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+/**
+ * Sequence of sorted elements.
+ * 
+ * Assumes each element may appear in the Seq at most once.
+ */
+trait StrictlySortedSeqLike[T] extends mutable.IndexedSeq[T] {
```

Comment:
If I wanted SortedArrayWrapper extends Seq with SortedSetLike, I need it to implement: contains, + (adding element) and -. Then I can override intersect, but:
```
def intersect(that: GenSet[A]): This
```
thus, if I override it, this method will be called for each GenSet not only SortedSets. This makes an implementation in SortedArrayWrapper not concise with the definition.